src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.PlaceholderLocal.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class NullableWalker
+    {
+        /// <summary>
+        /// A symbol to be used as a placeholder for an instance being constructed by
+        /// <see cref="BoundObjectCreationExpression"/>, or the input expression of a pattern-matching operation.
+        /// It is used to track the state of an expression, such as members being initialized.
+        /// </summary>
+        private sealed class PlaceholderLocal : LocalSymbol
+        {
+            private readonly Symbol _containingSymbol;
+            private readonly TypeWithAnnotations _type;
+            private readonly object _identifier;
+
+            public PlaceholderLocal(Symbol containingSymbol, object identifier, TypeWithAnnotations type)
+            {
+                Debug.Assert(identifier != null);
+                _containingSymbol = containingSymbol;
+                _type = type;
+                _identifier = identifier;
+            }
+
+            public override bool Equals(object obj)
+            {
+                if ((object)this == obj)
+                {
+                    return true;
+                }
+
+                var other = obj as PlaceholderLocal;
+
+                return (object)other != null && _identifier.Equals(other._identifier);
```

Comment:
Nit: could use `is` expression

Reply:
You mean `!(object is null)`?

Reply:
I mean `obj is Placeholderlocal other && _identifier.Equals(other._identifier)`

Reply:
OK, sure, but I did not write this code.  I mostly renamed the file.

---
In reply to: [268250124](https://github.com/dotnet/roslyn/pull/34273#discussion_r268250124) [](ancestors = 268250124)